lib/Client.js

```diff
@@ -2332,15 +2332,41 @@ class Client extends EventEmitter {
     /**
     * Get the ban list of a guild
     * @arg {String} guildID The ID of the guild
-    * @returns {Promise<Array<Object>>} Resolves with an array of {reason: String, user: User}
-    */
-    getGuildBans(guildID) {
-        return this.requestHandler.request("GET", Endpoints.GUILD_BANS(guildID), true).then((bans) => {
-            bans.forEach((ban) => {
-                ban.user = new User(ban.user, this);
-            });
-            return bans;
+    * @arg {Object} [options] Options for the request
+    * @arg {String} [options.after] Only get users after given user ID
+    * @arg {String} [options.before] Only get users before given user ID
+    * @arg {Number} [options.limit] The maximum number of users to return
+    * @returns {Promise<Array<Object>>} Resolves with an array of { reason: String, user: User }
+    */
+    async getGuildBans(guildID, options = {}) {
+        const bans = await this.requestHandler.request("GET", Endpoints.GUILD_BANS(guildID), true, {
+            after: options.after,
+            before: options.before,
+            limit: Math.min(options.limit, 1000) 
         });
+        
+        bans.forEach((ban) => {
```

Comment:
Change this to `map`

Reply:
Why should this be a map? We aren't returning a value

Reply:
`.map` will return an array with new objects. `.forEach` was used before and only updates the value of the property. An other option would be to use a for loop.

Reply:
Should also say move this logic to the return itself with the `.map`

Reply:
Would mess up the recursion.

Reply:
Resolved within discord. turns out i thought similar circumstance logic was different. whoops